src/Npgsql.NetTopologySuite/NetTopologySuiteHandler.cs

```diff
@@ -0,0 +1,348 @@
+﻿#region License
+// The PostgreSQL License
+//
+// Copyright (C) 2018 The Npgsql Development Team
+//
+// Permission to use, copy, modify, and distribute this software and its
+// documentation for any purpose, without fee, and without a written
+// agreement is hereby granted, provided that the above copyright notice
+// and this paragraph and the following two paragraphs appear in all copies.
+//
+// IN NO EVENT SHALL THE NPGSQL DEVELOPMENT TEAM BE LIABLE TO ANY PARTY
+// FOR DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES,
+// INCLUDING LOST PROFITS, ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS
+// DOCUMENTATION, EVEN IF THE NPGSQL DEVELOPMENT TEAM HAS BEEN ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//
+// THE NPGSQL DEVELOPMENT TEAM SPECIFICALLY DISCLAIMS ANY WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS
+// ON AN "AS IS" BASIS, AND THE NPGSQL DEVELOPMENT TEAM HAS NO OBLIGATIONS
+// TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+#endregion
+
+using GeoAPI.Geometries;
+using GeoAPI.IO;
+using NetTopologySuite.Geometries;
+using NetTopologySuite.IO;
+using Npgsql.BackendMessages;
+using Npgsql.TypeHandling;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+
+namespace Npgsql.NetTopologySuite
+{
+    public class NetTopologySuiteHandlerFactory : NpgsqlTypeHandlerFactory<IGeometry>
+    {
+        readonly IBinaryGeometryReader _reader;
+        readonly IBinaryGeometryWriter _writer;
+
+        internal NetTopologySuiteHandlerFactory(IBinaryGeometryReader reader, IBinaryGeometryWriter writer)
+        {
+            _reader = reader;
+            _writer = writer;
+        }
+
+        protected override NpgsqlTypeHandler<IGeometry> Create(NpgsqlConnection conn)
+            => new NetTopologySuiteHandler(_reader ?? new WKBReader(), _writer ?? new WKBWriter());
```

Comment:
Shouldn't we use the [the PostGIS-specific ones](https://github.com/NetTopologySuite/NetTopologySuite/tree/develop/NetTopologySuite.IO/NetTopologySuite.IO.PostGis2)? Any idea what PostGis2Reader does different/better than WKBReader? Maybe it supports the extra capabilities of PostGIS's EWKB or something?

Reply:
At first I thought about the `NetTopologySuite.IO.PostGis` package, but it doesn't support .NET Standard. Therefore, I decided to use classes from `NetTopologySuite`.

Replaced by `PostGisReader` and `PostGisWriter` from `NetTopologySuite.IO` which targetes to .NET Standard.

Reply:
Um, `PostGisReader` has an problem. It uses `Stream.Position` to set the current position ([source](https://github.com/NetTopologySuite/NetTopologySuite/blob/543b56b20c8c83df9d7ae12cf2ac3a15590c349f/NetTopologySuite.IO/NetTopologySuite.IO.PostGis/PostGisReader.cs#L100)) which isn't supported by `ColumnStream`. I will help them fix the issue.

Reply:
Interestingly, in their develop branch NetTopologySuite.IO.PostGis targets netstandard1.0, although NetTopologySuite.IO.PostGis2 is an old-style csproj which doesn't. Would you like to take this up with them, understand the differences between PostGis1, PostGis2 and the generic WKBReader/Writer, and also push them to do netstandard for PostGis2 if relevant?

This isn't blocking and can also be explored and fixed post-merge.

Reply:
Regarding the seeking (changing `Stream.Position`), remind them that PostGIS objects are theoretically streamed and cannot be assumed to be entirely in memory at any point...